### ISSUE TITLE
Need to pass timeout parameter to Patron adapter to avoid duplicate records in elasticsearch 

### ISSUE THREAD

**Issue by jet9** [uken/fluent-plugin-elasticsearch#50]
Default timeout value in /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/patron-0.4.18/lib/patron/session.rb is 5 seconds, but sometimes it's not enough to completely read bulk response from elasticseach and log data records sent to elasticsearch again and again. It leads to appearance a duplicate records in elasticsearch.

```
2014-07-01 16:35:23 +0400 [warn]: temporarily failed to flush the buffer. next_retry=2014-07-01 16:34:33 +0400 error_class="Faraday::TimeoutError" error="Operation timed out after 5000 milliseconds with 0 bytes received" instance=70263784889780
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/patron-0.4.18/lib/patron/session.rb:223:in `handle_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/patron-0.4.18/lib/patron/session.rb:223:in `request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/faraday-0.9.0/lib/faraday/adapter/patron.rb:33:in `call'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/faraday-0.9.0/lib/faraday/rack_builder.rb:139:in `build_response'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/faraday-0.9.0/lib/faraday/connection.rb:377:in `run_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-transport-1.0.2/lib/elasticsearch/transport/transport/http/faraday.rb:21:in `block in perform_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-transport-1.0.2/lib/elasticsearch/transport/transport/base.rb:187:in `call'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-transport-1.0.2/lib/elasticsearch/transport/transport/base.rb:187:in `perform_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-transport-1.0.2/lib/elasticsearch/transport/transport/http/faraday.rb:20:in `perform_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-transport-1.0.2/lib/elasticsearch/transport/client.rb:102:in `perform_request'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/elasticsearch-api-1.0.2/lib/elasticsearch/api/actions/bulk.rb:91:in `bulk'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-elasticsearch-0.3.0/lib/fluent/plugin/out_elasticsearch.rb:102:in `send'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluent-plugin-elasticsearch-0.3.0/lib/fluent/plugin/out_elasticsearch.rb:90:in `write'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.45/lib/fluent/buffer.rb:296:in `write_chunk'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.45/lib/fluent/buffer.rb:276:in `pop'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.45/lib/fluent/output.rb:310:in `try_flush'
  2014-07-01 16:35:23 +0400 [warn]: /usr/lib64/fluent/ruby/lib/ruby/gems/1.9.1/gems/fluentd-0.10.45/lib/fluent/output.rb:132:in `run'
```

**Comment by wangzhuzhen** [comment]
Is there any latest information about this?
I have the same duplicate records in EFK on k8s.

**Comment by Rocklviv** [comment]
Same issue with fluent 0.14.10
A lot of duplicated logs in EFK.

**Comment by pitr** [comment]
This issue was fixed with PR #59 which added `request_timeout` option
